### ISSUE TITLE
Compilation failure

### ISSUE THREAD

**Issue by coliss86** [MarginallyClever/Makelangelo-firmware#120]
**Describe the bug**
I cannot compile the project on the master branch. [The CI too](https://github.com/MarginallyClever/Makelangelo-firmware/runs/3246095083?check_suite_focus=true).

**To Reproduce**
Run `pio run` in the folder of the firmware and I get the following error : 
```
pio/libdeps/rumba-sixi3/TMCStepper/src/TMCStepper.h:25:11: fatal error: SoftwareSerial.h: No such file or directory

************************************************************************
* Looking for SoftwareSerial.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:SoftwareSerial.h"
* Web  > https://platformio.org/lib/search?query=header:SoftwareSerial.h
*
************************************************************************

  #include <SoftwareSerial.h>
           ^~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio/build/rumba-sixi3/libda9/TMCStepper/source/ENCMODE.cpp.o] Error 1
=============================================== [FAILED] Took 2.08 seconds ===============================================

Environment    Status    Duration
-------------  --------  ------------
rumba-sixi3    FAILED    00:00:02.076
========================================== 1 failed, 0 succeeded in 00:00:02.076 ==========================================
```

**Expected behavior**
The project compiles fine

**Additional context**
I made a first PR to build all environnement https://github.com/MarginallyClever/Makelangelo-firmware/pull/119

**Comment by i-make-robots** [comment]
should be fixed by df378efe8a87b95f6d6c2d7d711c0dc121b7e873.  can you confirm?

**Comment by i-make-robots** [comment]
I commented out inclusion of TMCStepper libraries after seeing that Marlin has the same issue, as of yet unresolved. https://github.com/teemuatlut/TMCStepper/issues/56

**Comment by coliss86** [comment]
Thanks !
It is getting better but [not all are OK](https://github.com/MarginallyClever/Makelangelo-firmware/runs/3326531149).

In https://github.com/MarginallyClever/Makelangelo-firmware/pull/119 I have enabled the compilation of all supported board, you can see what is wrong here: on the home page of the project, you will see a red cross : 
![image](https://user-images.githubusercontent.com/23615562/129447786-4eb50ce2-594a-48ee-badb-9393fa17d1a8.png)
then click on it : 
![image](https://user-images.githubusercontent.com/23615562/129447819-5685ea41-1c66-42ce-bb86-babb336e61e1.png)

Then you can click on the "details" on the line with red cross : 
![image](https://user-images.githubusercontent.com/23615562/129447870-9a31ffdb-8a6a-42a4-976b-7ef8f02e13d7.png)

You can also click on the "action" tab directly on the home page.

You'll have the log of what has been compile for each commit.

To sum up, the following boards still do not compile:
- BIGTREE_SKR_PRO-Sixi3
- BIGTREE_SKR_PRO-Stewart
- sixi2

**Comment by i-make-robots** [comment]
Yeah… every run sends me an email with results. Super fun. I’ll look at it in an hour.

**Comment by i-make-robots** [comment]
Ugh... making it work for STM is so.  much.  work.

**Comment by i-make-robots** [comment]
The HAL is basically copied from Marlin.  What M-F does differently is implement up to 6 axies and no extruder.

**Comment by i-make-robots** [comment]
STM is still open.  I guess that's my work for the week.
